### ISSUE TITLE
Crash far down in the stripe SDK when trying to add new card

### ISSUE THREAD

**Issue by cristeahub** [stripe/stripe-ios#1300]
## Summary
The Stripe SDK crashes when I try to add a card to my test account. The bottom of the trace stack is sdallocx before STDSSL_OPENSSL_free. Moving to the top of the stack a bit I can see the error happening here:

https://github.com/stripe/stripe-ios/blob/b17dd1265473a19b01b36dabc7e399359ea7bcff/Stripe/STPAPIRequest.m#L130

This again calls here:

https://github.com/stripe/stripe-ios/blob/59d4867eb84103afcb66be625deb5cc6a04d8282/Stripe/STPAPIClient.m#L786

Which calls here:
https://github.com/stripe/stripe-ios/blob/6d17243be30b3d4630d86d3771d3d61141dd91e1/Stripe/Payments/STPPaymentHandler.m#L434

After this there's a few more steps, but basically there's a lot of assembly before ending up at STDSSL_OPENSSL_free and sdallocx.

I'm not sure what to do to debug this further. I can consistently reproduce this crash.

## Code to reproduce

```objective-c
STPPaymentMethodCardParams *cardParams = [STPPaymentMethodCardParams new];
STPPaymentMethodBillingDetails *billingDetails = [STPPaymentMethodBillingDetails new];

cardParams.number = @"4000000000003220";
cardParams.expMonth = @12;
cardParams.expYear = @2023;
cardParams.cvc = @"123";

billingDetails.email = @"yo@yo.com";
billingDetails.name = @"Navn Navnesen";

STPPaymentMethodParams *paymentMethodParams = [STPPaymentMethodParams paramsWithCard:cardParams
                                                                      billingDetails:billingDetails
                                                                            metadata:@{
                                                                                       @"user_id": @1
                                                                                       }];

[[STPAPIClient sharedClient] createPaymentMethodWithParams:paymentMethodParams
                                                completion:^(STPPaymentMethod * _Nullable paymentMethod, NSError * _Nullable error)
{
  STPSetupIntentConfirmParams *setupIntentParams = [[STPSetupIntentConfirmParams alloc] initWithClientSecret:@"<secret from setup intent>"];
  setupIntentParams.paymentMethodID = paymentMethod.stripeId;
  setupIntentParams.returnURL = @"https://<callbackurl>";
  [[STPPaymentHandler sharedHandler] confirmSetupIntent:setupIntentParams
                              withAuthenticationContext:self
                                             completion:^(STPPaymentHandlerActionStatus handlerStatus, STPSetupIntent *setupIntent, NSError *error)
    {
      NSLog(@"Status: %ld", (long)handlerStatus);
    }];
}];
```

## iOS version
12.4

## Installation method
CocaPods

## SDK version
16.0.5

## Other information
I haven't used this lib a lot, so I'm still in the process of making the code better, however this code did work at some point before it started crashing all the time.

**Comment by davidme-stripe** [comment]
This could be caused by a conflict with a copy of OpenSSL in another CocoaPod. We added a fix to yesterday’s 16.0.6 — try updating and let us know if it resolves the issue.

**Comment by cristeahub** [comment]
Thanks, it seems like https://github.com/stripe/stripe-ios/pull/1297 indeed did fix this.
